HDDS-11944. Usability improvements for Ozone tools.

errose28 opened the pull request:
## What changes were proposed in this pull request?

Document a proposal for a new layout of the `ozone debug` and `ozone repair` CLIs as part of the usability improvements in HDDS-11944.

## What is the link to the Apache JIRA

HDDS-11944

## How was this patch tested?

N/A

errose28 commented:
cc @Tejaskriya @sarvekshayr @ptlrs

adoroszlai reviewed (approved):
Thanks @errose28 for the proposal, LGTM.

adoroszlai commented on hadoop-hdds/docs/content/design/tools.md:
```diff
@@ -0,0 +1,136 @@
+
+- `ozone repair om`
+    These commands make repairs to an individual OM instance.
+    - `ozone repair om quota`
+        The new location of `ozone repair ldb quota`
```
`ozone repair ldb quota` does not exist currently, but I found this:

```
Usage: ozone repair quota [COMMAND]
Operational tool to repair quota in OM DB.
Commands:
  status  CLI to get the status of last trigger quota repair if available.
  start   CLI to trigger quota repair.
```

which is an online command (requires running OM).

adoroszlai reviewed (changes requested):
Found a mismatch while working on the `--dry-run` flag.

ivandika3 commented:
HDDS-11944 is well underway, @errose28 is this ready to be merged?

errose28 commented:
Thanks for reviewing @dombizita I've updated the doc with your suggestions to make it match what is currently on master. I've also added a disclaimer at the top that the doc can serve as a reference going forward but that it is not expected to be up to date with the current state of the CLI.

dombizita commented on hadoop-hdds/docs/content/design/tools.md:
```diff
@@ -0,0 +1,145 @@
+    Currently all online debug tools fit under this category, where they get a listing from the OM and perform checks on the replica information retrieved. The help message for this subcommand should explicitly state that it runs over the network against a running cluster.
+    - `ozone debug replicas chunk-info`
+        This is the new location of `ozone debug chunkinfo`
+    - `ozone debug replicas verify [--checksums] [--blocks]`
```
```suggestion
    - `ozone debug replicas verify [--checksums] [--block-existence] [--container-state]`
```

dombizita commented on hadoop-hdds/docs/content/design/tools.md:
```diff
@@ -0,0 +1,145 @@
+        A dedicated subcommand for all certificate related repairs on SCM we may need now or in the future.
+        - `ozone repair scm cert recover`
+            The new location of `ozone repair cert-recover`
+- `ozone repair scm`
```
This is supposed to be `ozone repair datanode`, right?
```suggestion
- `ozone repair datanode`
```

dombizita reviewed (commented):
Thank you for updating it @errose28! I have a few small comments, otherwise it looks good to me!

errose28 commented:
Good catches, just applied the updates @dombizita.

errose28 commented:
Thanks all for reviewing and helping with this project!
